### PR TITLE
Docs/pull request template

### PR DESCRIPTION
**Pull request by blackboxsw**
Add a doc template for github which will require folks do the following.

## Proposed Commit Message

notices: clear lock file and notice when encountering any exception

When the commandline encounters an unexpected error, the                        
main_error_handler needs to call UAConfig.delete_cache_key("lock")              
to ensure any Operation in progress notices messages are also                   
removed. This most often happens during auto-attach on PRO machines             
because it raises a UserFacingError saying:                                     
                                                                                
This machine is already attached to <some-account>                              
                                                                                
Since main_error_handler doesn't have direct access to an                       
instantiated UAConfig, we store that UAConfig.delte_cache_key                   
callable on _CLEAR_LOCK_FILE                                                    
                                                                                
Fixes: #1326

## Test Steps

* Launch an Ubuntu PRO Machine with this branch
```bash
UACLIENT_BEHAVE_BUILD_PR=1 UACLIENT_BEHAVE_DESTROY_INSTANCES=0 
UACLIENT_BEHAVE_AWS_ACCESS_KEY_ID=<YOUR_AWS_ACCESS_KEY_ID>  UACLIENT_BEHAVE_AWS_SECRET_ACCESS_KEY=<YOUR_AWS_SECRET_ACCESS_KEY> tox -e behave-awspro-18.04 
--no-capture
```
* That run should leave around a test PRO bionic instance lying around after test, you can ssh to it with your ec2-uaclient-ci-0127-dev-0126-220857.pem in the local directory
`ls -ltr ec2*pem`
* Reboot the machine and run status afterward to confirm no "Operation in progress: auto-attach" message 
```bash
ssh -i ec2.....pem ubuntu@<PRO_IP_ADDRESS>  sudo reboot
ssh -i ec2.....pem ubuntu@<PRO_IP_ADDRESS>  sudo ua status

```
## Desired commit type::

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+## Proposed Commit Message
+<!-- Include a proposed commit message because all PRs can be merged in a variety of ways by the reviewer -->
+
+> summary: no more than 70 characters
+>
+> A description of what the change being made is and why it is being
```
Comment (mitechie) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: Requirements should be a bullet list. 

- what is the change?
- why is it made?
- a blank line is required
- wrap at 72 chars, if your text is wider than this sentence then please wrap moar
- use multiple paragraphs as appropriate

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+## Proposed Commit Message
+<!-- Include a proposed commit message because all PRs can be merged in a variety of ways by the reviewer -->
+
+> summary: no more than 70 characters
```
Comment (mitechie) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: summary: which should be no wider than this length here which is at 70chars

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+>
+> A description of what the change being made is and why it is being
+> made, if the summary line is insufficient.  The blank line above is
+> required. This should be wrapped at 72 characters, but otherwise has
+> no particular length requirements.
+>
+> If you need to write multiple paragraphs, feel free.
+>
+> LP: #NNNNNNN (replace with the appropriate Launchpad bug reference if applicable)
```
Comment (mitechie) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: #BugNumberHere

**Review by mitechie — COMMENTED** [review]
ty, some suggestions

**Review by lucasmoura — COMMENTED** [review]
I think we have code here from #1339

After we remove that, I think we can merge this.

**Review by mitechie — APPROVED** [review]

**Pull request merged**